src/linear_algebra/special_linear_group.lean

```diff
@@ -144,6 +154,22 @@ def to_GL : special_linear_group n R →* general_linear_group R (n → R) :=
 
 lemma coe_to_GL (A : special_linear_group n R) : ↑A.to_GL = A.to_lin'.to_linear_map := rfl
 
+section has_neg
+
```

Comment:
I think it would make sense to add a `variables` line for this section:
```
variables {R : Type*} [comm_ring R] [fact (even (fintype.card n))]
```

Reply:
We already have the variables 
```
variables (R : Type v) [comm_ring R]
```
at the top of the file, so this will conflict.

Reply:
Then we don't even need to redeclare R, right?
```suggestion
variables [fact (even (fintype.card n))]

```

Reply:
You're right, the `{R : Type*} [comm_ring R]` can go from both.  But if we make a variables line for the `fact` as you propose, it's no longer possible to name it `_i` in the first lemma and get access to `_i.elim`.

Reply:
I think you can just use `fact.out (even (fintype.card n))` instead of `i.elim` which avoids having to name it. You might even be able to just use `fact.out _`.

Reply:
Ah nice, I didn't know that syntax.